fenix/app/src/main/java/org/mozilla/fenix/debugsettings/data/DebugSettingsCache.kt

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.debugsettings.data
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+
+/**
+ * [DataStore] for accessing debugging settings.
+ */
+private val Context.debugSettings: DataStore<Preferences> by preferencesDataStore(name = "debug_settings")
+
+private val debugDrawerEnabledKey = booleanPreferencesKey("debug_drawer_enabled")
+
+/**
+ * Cache for accessing any settings related to debugging.
+ */
+interface DebugSettingsCache {
+
+    /**
+     * [Flow] for checking whether the Debug Drawer is enabled.
+     */
+    val debugDrawerEnabledFlow: Flow<Boolean>
+
+    /**
+     * Updates whether the debug drawer is enabled.
+     *
+     * @param enabled Whether the debug drawer is enabled.
```

Comment:
When the other PR is merged, it'd be nice to update mentions of `debug drawer` here in comments with `[DebugDrawer]`